tests/test_cases/issue_1031/issue_1031.py

```diff
@@ -0,0 +1,24 @@
+from cocotb import test, fork
+from cocotb.triggers import Lock, Timer
+
+@test()
+async def test_trigger_lock(dut):
+
+    resource = [0]
+    lock = Lock()
+
```

Comment:
Would be easier to follow if you defined this within the test
```suggestion

    async def co():
        nonlocal resource
        await Timer(10, "ns")
        async with lock:
            for i in range(4):
                resource += 1
            await Timer(10, "ns")
```